### PR TITLE
Apply filters to a Hugging Face dataset to avoid repeating all variants.

### PR DESCRIPTION
**Pull request by marcenacp**
The only issue for now is that `regex` is a regex while `includes` is a glob... So I use heuristics to convert from one to another. I think it's not a problem for hugging face datasets as we control the form they have. But it can be challenging to have a generic conversion. The best would be to use either regular expressions or glob patterns everywhere.

### CODE REVIEW

File: python/mlcroissant/mlcroissant/_src/datasets_test.py [hunk 1]
```diff
@@ -192,13 +191,16 @@ def test_nonhermetic_loading(version, dataset_name, record_set_name, num_records
 # Non-hermetic test cases for croissant >=1.0 only (data from the internet).
 @pytest.mark.nonhermetic
 @pytest.mark.parametrize(
-    ["dataset_name", "record_set_name", "num_records"],
+    ["dataset_name", "record_set_name", "num_records", "filters"],
```
Comment (ccl-core) [inline python/mlcroissant/mlcroissant/_src/datasets_test.py:194]: So here we could also e.g. have 

`["huggingface-anthropic-hh-rlhf/metadata.json", "harmless-base", 10, "train"]`

right?

**Review by ccl-core — APPROVED** [review]
That's cool! Thanks :)

File: python/mlcroissant/mlcroissant/_src/datasets_test.py (hunk 1 above)
Comment (marcenacp) [reply python/mlcroissant/mlcroissant/_src/datasets_test.py:194]: I haven't tried this dataset yet!

Comment (marcenacp) [reply python/mlcroissant/mlcroissant/_src/datasets_test.py:194]: But potentially yes.

**Pull request merged**